src/main/angular/app/date_select/date_select.component.ts

```diff
@@ -0,0 +1,90 @@
+import {
+  Component,
+  OnInit,
+  Input,
+  SimpleChanges,
+  Output,
+  EventEmitter,
+} from '@angular/core';
+import {DateUtilitiesService} from '../services/date_utilities.service';
+import {DateRange} from '../../model/date_range';
+
+/** Used for selecting the dates to display on the graph */
+@Component({
+  selector: 'app-date-select',
+  templateUrl: './date_select.component.html',
+  styleUrls: ['./date_select.component.css'],
+})
+export class DateSelectComponent implements OnInit {
+  /** The possible range for the date select. */
+  @Input()
+  public possibleRange: DateRange;
+
+  /** Sends the range selected by the user back out when it changes. */
+  @Output()
+  public selectedRange = new EventEmitter<DateRange>();
+
+  /** The DataRange that was sent last. */
+  public lastSent: DateRange;
+
+  public currentMin: Date;
+  public currentMax: Date;
+
+  constructor(private dateUtilities: DateUtilitiesService) {}
+
+  ngOnInit(): void {}
+
+  /** Called when an input field changes. */
+  ngOnChanges(changes: SimpleChanges) {
```

Comment:
could you set a quick time to discuss this design?

It's not clear from the impl exactly how many changes will be triggered on a single value change elsewhere in the application.

Reply:
Of course!